platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java

```diff
@@ -187,6 +212,16 @@ void setZoom(double zoom, @NonNull PointF focalPoint) {
   }
 
   void setZoom(double zoom, @NonNull PointF focalPoint, long duration) {
+    mapView.addOnMapChangedListener(new MapView.OnMapChangedListener() {
+      @Override
+      public void onMapChanged(int change) {
+        Timber.e("CHANGE : %s", change);
+        if (change == MapView.REGION_DID_CHANGE_ANIMATED) {
+          mapView.removeOnMapChangedListener(this);
```

Comment:
Wondering if it order matters 🤔 
In `setZoom` is defined `mapView.removeOnMapChangedListener` and `cameraChangeDispatcher.onCameraIdle` second but in `moveBy` is defined the other way around.